src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/Services/IdentityAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,59 @@
+using BlazorWasm_CSharp.Shared.Models;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace BlazorWasm_CSharp.Client
+{
+    public class IdentityAuthenticationStateProvider : AuthenticationStateProvider
```

Comment:
Since the client is agnostic to what technology is being used on the server (i.e., it doesn't care whether it's Identity or something else), could we name this to `ServerAuthenticationStateProvider` or similar?

Reply:
Do you have any other suggestion? I see this as an extension of Identity on the app if you ask me. I think `ServerAuthenticationStateProvider` is a bad choice given the fact that there is already a `ServerAuthenticationStateProvider` class in ASP.NET Core, but I'm open to suggestions.

Reply:
Hmm, maybe `HostAuthenticationStateProvider`, since it gets state from the "host" app in the hosted template scenario?